consensus/replay.go

```diff
@@ -259,12 +259,9 @@ func (h *Handshaker) Handshake(proxyApp proxy.AppConns) error {
 		"protocol-version", res.AppVersion,
 	)
 
-	// Set AppVersion on the state.
-	if h.initialState.Version.Consensus.App != version.Protocol(res.AppVersion) {
+	if h.initialState.LastBlockHeight == 0 { //we only update state when we are in initial state
 		h.initialState.Version.Consensus.App = version.Protocol(res.AppVersion)
```

Comment:
Why do we need to update version here? can't we do it in ReplayBlocks when `if stateBlockHeight == 0`

Reply:
If we don't, the version in the state mismatches with the app version, the app version is supposed to be returned in EndBlock